### DISCUSSION TITLE
规则界面获取不到app图标，提示读取应用列表权限，其实有权限，但是仍然提示

### DISCUSSION THREAD

**Discussion by dearmb** [gkd-kit/gkd#796] [General]
![GKD](https://github.com/user-attachments/assets/0dd48d8c-8d6e-4305-aaab-f6af61ee2011)
规则界面获取不到app图标，提示读取应用列表权限，其实有权限，但是仍然提示

**Comment by lisonge** [comment]
系统问题，我不知道怎么回事，也不知道怎么解决

**Reply by dearmb** [reply]
我怎么感觉没升级之前是没问题的呀
昨天好像升级了

**Reply by lisonge** [reply]
你说的这些我不知道，应该是你的系统的问题，我这边无法复现你的问题

如果你会 Android 开发，可以尝试解决此问题，欢迎提关于此问题的 pr

**Reply by dearmb** [reply]
> 你说的这些我不知道，应该是你的系统的问题，我这边无法复现你的问题
> 
> 如果你会 Android 开发，可以尝试解决此问题，欢迎提关于此问题的 pr

我不会，我觉得是升级的问题，我备份了我的规则，重新降级安装到了1.9.1，app图标也好使了，刚刚说的迅雷新加的规则也好使了

**Reply by lisonge** [reply]
不点击跟规则逻辑没关系

单纯是因为 gkd 检测不到 迅雷 安装，所以没有启用迅雷的规则

总之就是应用列表权限的问题

我猜测是你通过因为 gkd 内部更新后，系统判断安装来源是 gkd ，就给了一个假权限

你重新下载最新安装包后，使用系统自带文件管理器安装试试

**Reply by dearmb** [reply]
> 不点击跟规则逻辑没关系
> 
> 单纯是因为 gkd 检测不到 迅雷 安装，所以没有启用迅雷的规则
> 
> 总之就是应用列表权限的问题
> 
> 我猜测是你通过因为 gkd 内部更新后，系统判断安装来源是 gkd ，就给了一个假权限
> 
> 你重新下载最新安装包后，使用系统自带文件管理器安装试试

不知道系统什么问题，我现在还是miui，好几年也没更新，不知道系统什么问题，我现在还是miui，好几年也没更新，直接更新用的安装不是调用“系统自带文件管理器”安装的吗？
而且检测不到迅雷的安装的话，为什么开屏广告没问题？

**Reply by lisonge** [reply]
开屏广告是全局规则，不受到此影响

是调用系统 apk 安装器去安装，不是文件管理器

**Reply by dearmb** [reply]
没找到你所说的系统apk安装器，我也不是新用户，之前升级那么多次，安装方法全都一模一样，这次版本升级就不好使，不理解为什么是我系统的问题，系统也两年没升级，不行继续用1.9.1，看以后的新版本会不会还这样吧

**Reply by lisonge** [reply]
你没懂我的意思，所有的安装都通过系统安装器

但是总得有另外一个应用去发起请求告诉安装器去安装新应用

关键在于发起这个安装请求的应用是不是系统认可的安全来源应用

**Reply by lisonge** [reply]
算了别管这个问题了

你的问题我无法复现，也无法解决

能用就将就用，用不了就自己编译安装或者换其他的
